docs/source/integrations/mlflow.md

```diff
@@ -0,0 +1,311 @@
+
+```{image} ../meta/images/mlflow-artifact-preview-image.png
+:alt: MLflow image preview thanks to the artifact tracking capabilities of kedro-mlflow
+:width: 80%
+:align: center
+```
+
+:::{warning}
+If you get a `Failed while saving data to data set MlflowMatplotlibWriter` error,
+it's probably because you had already executed `kedro run` while the dataset was marked as `versioned: true`.
+The solution is to cleanup the old `data/08_reporting/dummy_confusion_matrix.png` directory.
+:::
+
+Check out {external+kedro-mlflow:doc}`the official kedro-mlflow documentation on versioning Kedro datasets <source/04_experimentation_tracking/03_version_datasets>`
+for more information.
+
+### Model registry in MLflow using `kedro-mlflow`
+
+If your Kedro pipeline trains a machine learning model, you can track those models in MLflow
+so that you can manage and deploy them.
+The `kedro-mlflow` plugin introduces a special artifact, `MlflowModelTrackingDataset`,
+that you can use to load and save your models as MLflow artifacts.
+
+For example, if you have a dataset corresponding to a scikit-learn model,
+you can modify it as follows:
+
+```diff
+ regressor:
+-  type: pickle.PickleDataset
+-  filepath: data/06_models/regressor.pickle
+-  versioned: true
++  type: kedro_mlflow.io.models.MlflowModelTrackingDataset
++  flavor: mlflow.sklearn
+```
+
+The `kedro-mlflow` Hook will log the model as part of the run
+in {external+mlflow:doc}`the standard MLflow Model format <models>`.
+
+If you also want to _register_ it
+(hence store it in the MLflow Model Registry)
+you can add a `registered_model_name` parameter:
+
+```{code-block} yaml
+:emphasize-lines: 4-5
+
+regressor:
+  type: kedro_mlflow.io.models.MlflowModelTrackingDataset
+  flavor: mlflow.sklearn
+  save_args:
+    registered_model_name: spaceflights-regressor
+```
+
+Then you will see it listed as a Registered Model:
+
+```{image} ../meta/images/kedro-mlflow-registered-model.png
+:alt: MLflow Model Registry listing one model registered with kedro-mlflow
+:width: 80%
+:align: center
+```
+
+To load a model from a specific run, you can specify the `run_id`.
+For that, you can make use of {ref}`runtime parameters <runtime-params>`:
+
+```{code-block} yaml
+:emphasize-lines: 13
+
+# Add the intermediate datasets to run only the inference
+X_test:
+  type: pandas.ParquetDataset
+  filepath: data/05_model_input/X_test.pq
+
+y_test:
+  type: pandas.CSVDataset  # https://github.com/pandas-dev/pandas/issues/54638
+  filepath: data/05_model_input/y_test.csv
+
+regressor:
+  type: kedro_mlflow.io.models.MlflowModelTrackingDataset
+  flavor: mlflow.sklearn
+  run_id: ${runtime_params:mlflow_run_id,null}
+  save_args:
+    registered_model_name: spaceflights-regressor
+```
+
+And specify the MLflow run id on the command line as follows:
+
+```
+$ kedro run --to-outputs=X_test,y_test
+...
+$ kedro run --from-nodes=evaluate_model_node --params mlflow_run_id=4cba84...
```

Comment:
This was somewhat clunky, dumped some thoughts in https://github.com/kedro-org/kedro/discussions/3922

Reply:
I agree, but it's great that you added that section! It's an interesting functionality.